src/test/java/com/google/finapp/FinAppIT.java

```diff
@@ -19,68 +19,104 @@
+import com.google.cloud.spanner.Database;
 import com.google.cloud.spanner.DatabaseClient;
-import com.google.cloud.spanner.DatabaseId;
+import com.google.cloud.spanner.IntegrationTest;
+import com.google.cloud.spanner.IntegrationTestEnv;
 import com.google.cloud.spanner.Key;
 import com.google.cloud.spanner.KeySet;
 import com.google.cloud.spanner.ResultSet;
-import com.google.cloud.spanner.Spanner;
-import com.google.cloud.spanner.SpannerOptions;
+import com.google.cloud.spanner.testing.RemoteSpannerHelper;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
 import java.math.BigDecimal;
 import java.util.Arrays;
+import java.util.List;
+import java.util.Scanner;
 import java.util.UUID;
-import org.junit.jupiter.params.ParameterizedTest;
-import org.junit.jupiter.params.provider.ValueSource;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
 
+@Category(IntegrationTest.class)
 public class FinAppIT {
 
   private static SpannerDaoInterface JDBCDao;
   private static SpannerDaoInterface JavaDao;
   private static DatabaseClient databaseClient;
+  private static Database db;
 
-  static {
-    // set all IDs according to emulator setup
-    String spannerProjectId = "test-project";
-    String spannerInstanceId = "test-instance";
-    String spannerDatabaseId = "test-database";
-    JDBCDao = new SpannerDaoJDBCImpl(spannerProjectId, spannerInstanceId, spannerDatabaseId);
-    SpannerOptions spannerOptions = SpannerOptions.getDefaultInstance();
-    Spanner spanner = spannerOptions.toBuilder().build().getService();
-    databaseClient = spanner
-        .getDatabaseClient(DatabaseId.of(spannerProjectId, spannerInstanceId, spannerDatabaseId));
+  @ClassRule
+  public static IntegrationTestEnv env = new IntegrationTestEnv();
+
+  @BeforeClass
+  public static void setup() {
+    final RemoteSpannerHelper testHelper = env.getTestHelper();
+    try {
+      db = testHelper
+          .createTestDatabase(extractStatementsFromSDLFile("src/test/resources/schema.sdl"));
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+    final String databaseId = db.getId().getDatabase();
+    final String projectId = testHelper.getOptions().getProjectId();
+    final String instanceId = testHelper.getInstanceId().getInstance();
+    JDBCDao = new SpannerDaoJDBCImpl(projectId, instanceId, databaseId);
+    databaseClient = testHelper.getDatabaseClient(db);
     JavaDao = new SpannerDaoImpl(databaseClient);
   }
 
-  private SpannerDaoInterface getImpl(String impl) {
-    if (impl.equals("java client")) {
-      return JavaDao;
-    } else {
-      return JDBCDao;
+  @AfterClass
+  public static void tearDown() {
+    db.drop();
+  }
+
+  @Test
+  public void createAccountTest() throws SpannerDaoException {
```

Comment:
A test's name should summarize the behavior it is testing. A good name describes both the actions that are being taken on a system and the expected outcome. It can get pretty verbose and that's ok.